src/State/Hub.php

```diff
@@ -229,17 +233,9 @@ public static function setCurrent(HubInterface $hub): HubInterface
         return $hub;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getIntegration(string $className): ?IntegrationInterface
+    public function addIntegration(IntegrationInterface $integration): void
     {
-        $client = $this->getClient();
-        if (null !== $client) {
-            return $client->getIntegration($className);
-        }
-
-        return null;
+        $this->integrations[] = $integration->bindToHub($this);
```

Comment:
I don't like the fact that an integration could be bounded later to a different hub. Imho if this is the direction we are going to follow then an integration should require the hub to bind in the constructor